### PR TITLE
Fixed case of cy8c6xxa target in board ID list

### PR DESCRIPTION
**Pull request by mcgordonite**
This fixes the `is_target_supported` property of `pyocd json --boards` for the CY8CPROTO-062-4343W.

Before:
```json
{
  "id": "1901",
  "name": "CY8CPROTO-062-4343W",
  "target": "cy8c6xxA",
  "binary": "l1_cy8c6xxa.bin",
  "is_target_builtin": false,
  "is_target_supported": false
}
```

After, `is_target_supported` is `true`:
```json
{
  "id": "1901",
  "name": "CY8CPROTO-062-4343W",
  "target": "cy8c6xxa",
  "binary": "l1_cy8c6xxa.bin",
  "is_target_builtin": true,
  "is_target_supported": true
}
```

### CODE REVIEW

**Comment by arekzaluski** [comment]
@flit  @0xc0170  Can you please take a look at it? Mbed Studio started using the `is_target_supported` field in version 1.2.0 (released last week).

**Comment by flit** [comment]
It really should be using a case-insensitive comparison, since the comparison for the `--target` argument is case-insensitive.

File: pyocd/board/board_ids.py [hunk 1]
```diff
@@ -195,24 +195,24 @@ def __init__(self, name, target, binary):
...
     "1903": BoardInfo(  "CYW9P62S1-43012EVB-01","cy8c6xx7_s25fs512s", "l1_cy8c6xx7.bin",    ),
     "1904": BoardInfo(  "CY8CPROTO-063-BLE",    "cy8c6xx7_nosmif",  "l1_cy8c6xx7.bin",      ),
-    "1905": BoardInfo(  "CY8CKIT-062-4343W",    "cy8c6xxA",         "l1_cy8c6xxa.bin",      ),
+    "1905": BoardInfo(  "CY8CKIT-062-4343W",    "cy8c6xxa",         "l1_cy8c6xxa.bin",      ),
     "1906": BoardInfo(  "CYW943012P6EVB-01",    "cy8c6xx7",         "l1_cy8c6xx7.bin",      ),
     "1907": BoardInfo(  "CY8CPROTO-064-SB",     "cy8c64xx_cm4_s25hx512t", "l1_cy8c6xx7.bin",),
     "1908": BoardInfo(  "CYW9P62S1-43438EVB-01","cy8c6xx7",         "l1_cy8c6xx7.bin",      ),
-    "1909": BoardInfo(  "CY8CPROTO-062S2-43012","cy8c6xxA",         "l1_cy8c6xxa.bin",      ),
+    "1909": BoardInfo(  "CY8CPROTO-062S2-43012","cy8c6xxa",         "l1_cy8c6xxa.bin",      ),
     "190A": BoardInfo(  "CY8CKIT-064S2-4343W",  "cy8c64xA_cm4",     "l1_cy8c6xxa.bin",      ),
```
Comment (arekzaluski) [inline pyocd/board/board_ids.py:207]: This one was missed. I agree with @flit. It is better to modify the code to be case-insensitive.

**Comment by flit** [comment]
(test failure was an intermittent `gdb_test.py` failure unrelated to this code.)

File: pyocd/tools/lists.py [hunk 2]
```diff
@@ -89,7 +89,10 @@ def list_boards(name_filter=None):
             'boards' : boards
             }
 
+        # Lowercase target names for comparison
         managed_targets = [dev.part_number.lower() for dev in pack_target.ManagedPacks.get_installed_targets()]
+        builtin_target_names = [target_name.lower() for target_name in BUILTIN_TARGETS]
+        target_names = [target_name.lower() for target_name in TARGET]
```
Comment (mcgordonite) [inline pyocd/tools/lists.py:95]: @flit not sure if I've gone too far here: I'm converting the target name from the target list to lower case in addition to the target from the board info. The bug in this case was isolated to the board info list.

Comment (flit) [reply pyocd/tools/lists.py:95]: No, I think this is perfect.

**Review by flit — APPROVED** [review]

**Pull request merged**